CMakeLists.txt

```diff
@@ -16,6 +16,10 @@ if(NOT "${CMAKE_VERSION}" VERSION_LESS "3.27")
   cmake_policy(SET CMP0145 OLD)
 endif()
 
+# CXX STANDARD
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
```

Comment:
Can we set this on the qgis_core target instead of globally ([`set_target`](https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html)) so it's properly propagated also to downstream apps (I have e.g. QField in mind here)

Reply:
With this it would maybe also work to have different standards for the python binding targets ?!